operator/api/v1beta1/helmrelease_types.go

```diff
@@ -69,12 +69,12 @@ type HelmReleaseCondition struct {
 // ChartSpec defines the desired Helm chart
 type ChartSpec struct {
 	Repository string `json:"repository"`
-	Path       string `json:"path"`
-	Revision   string `json:"revision"`
+	Name       string `json:"name"`
+	Version    string `json:"version"`
 }
 
-func (cs *ChartSpec) URI() string {
-	return path.Join(cs.Repository, cs.Path)
+func (c ChartSpec) URL() string {
+	return fmt.Sprintf("%s/%s", c.Repository, c.Name)
```

Comment:
We could `TrimSuffix(c.Repository, "/")` to handle an edge case like `s3://helm.wattpad.com/`. Since the chart now has a `Name` instead of `Path` i don't think we would have to worry about leading/trailing slashes there.